code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+	var/obj/structure/barricade/cade = /obj/structure/barricade
+	if(is_blocked_turf(buildplace))
+		for(var/obj/thing in buildplace)
+			if(istype(thing, cade) && thing.dir != fobdrone.dir)
+				break
+			else
+				to_chat(owner, "<span class='warning'>No space here for a barricade.</span>")
+				return
+	if(!do_after(fobdrone, 3 SECONDS, FALSE, buildplace, BUSY_ICON_BUILD))
+		return
+	console.metal_remaining -= 4
+	cade = new /obj/structure/barricade/metal(buildplace)
+	cade.setDir(fobdrone.dir)
+	if(console.do_wiring)
+		if(console.metal_remaining <= 1)
+			to_chat(owner, "<span class='warning'>Not enough material for razor-wiring.</span>")
+			return
+
+		console.metal_remaining -=2
+		cade.wire()
+
+
+/datum/action/innate/remote_fob/plast_cade
+	name = "Place Plasteel Barricade"
+	action_icon_state = "plast_cade"
+
+/datum/action/innate/remote_fob/plast_cade/Activate()
+	. = ..()
+	if(. || !check_spot())
+		return
+
+	if(console.plasteel_remaining < 5)
+		to_chat(owner, "<span class='warning'>Out of material.</span>")
+		return
+
+	var/turf/buildplace = get_turf(fobdrone)
+	var/obj/structure/barricade/cade = /obj/structure/barricade
+	if(is_blocked_turf(buildplace))
+		for(var/obj/thing in buildplace)
+			if(istype(thing, cade) && thing.dir != fobdrone.dir)
+				break
+			else
+				to_chat(owner, "<span class='warning'>No space here for a barricade.</span>")
+				return
+	if(!do_after(fobdrone, 3 SECONDS, FALSE, buildplace, BUSY_ICON_BUILD))
+		return
+	console.plasteel_remaining -= 5
+	cade = new /obj/structure/barricade/plasteel(buildplace)
+	cade.setDir(fobdrone.dir)
+	cade.closed = FALSE
+	cade.density = TRUE
+	cade.update_icon()
+	cade.update_overlay()
+	if(console.do_wiring)
+		if(console.metal_remaining <= 1)
+			to_chat(owner, "<span class='warning'>Not enough material for razor-wiring.</span>")
+			return
+
+		console.metal_remaining -=2
+		cade.wire()
+		
+
+/datum/action/innate/remote_fob/toggle_wiring
+	name = "Toggle Razorwire"
+	action_icon_state = "wire"
+
+/datum/action/innate/remote_fob/toggle_wiring/Activate()
+	. = ..()
+	if(.)
+		return
+	console.do_wiring = !console.do_wiring
+	to_chat(owner, "<span class='notice'>Will now [console.do_wiring ? "do wiring" : "stop wiring"].</span>")
+
+/datum/action/innate/remote_fob/sentry
+	name = "Place Sentry"
+	action_icon_state = "sentry"
+
+/datum/action/innate/remote_fob/sentry/Activate()
+	. = ..()
+	if(. || !check_spot())
+		return
+	var/obj/machinery/marine_turret/turret = /obj/machinery/marine_turret
+	var/turf/buildplace = get_turf(fobdrone)
+	var/obj/structure/barricade/cade = /obj/structure/barricade
+	if(console.sentry_remaining < 1)
+		to_chat(owner, "<span class='warning'>You need to redeem a Sentry voucher to place one.</span>")
+		return
+	if(is_blocked_turf(buildplace))
+		for(var/obj/thing in buildplace)
+			if(istype(thing, cade))
+				break
+			else
+				to_chat(owner, "<span class='warning'>No space here for a sentry.</span>")
+				return
```

Comment:
I see this repeated 3 times. Maybe make a `remote_fob` proc to avoid repeated code?

Reply:
the conditions are different for the sentry. sentries can be placed on any nonblocked turfs, aswell as turfs which are blocked exclusively by barricades

Reply:
Ah, alright.